### DISCUSSION TITLE
Add venv as an environment manager option

### DISCUSSION THREAD

**Discussion by nickcorona** [drivendataorg/cookiecutter-data-science#381] [Requested tools]
It's the default python virtual environment library now and ships with python 3.

**Comment by floer32** [comment]
Good call. 

various tools auto-discover “venv” and “.venv” - including IDEs like PyCharm and the (wonderful) package manager, Poetry

**Comment by pjbull** [comment]
We don't use `venv` in our projects could one of you (@hangtwenty @nickcorona) point towards a resource with the basics so that it can be implemented with #162?

**Comment by floer32** [comment]
Hah I was misunderstanding this thread. I thought it was talking about which directory name to use for a virtualenv if created within project. But I think OP was talking about venv vs virtualenv libraries in stdlib


________________________________

**Comment by nickcorona** [comment]
> Hah I was misunderstanding this thread. I thought it was talking about which directory name to use for a virtualenv if created within project. But I think OP was talking about venv vs virtualenv libraries in stdlib

Yeah, that is what I mean.

**Comment by jamesmyatt** [comment]
The general advice is to still prefer `virtualenv` since it's faster, has more features and can be upgraded via pip.

_Update_: In most cases, "virtualenv" is still recommended over just "venv". See the https://virtualenv.pypa.io/en/stable/index.html for details, but basically (as I understand it), only the minimum functionality was migrated to venv to enable everyone to get basic native isolated environments without pip-installing anything. So if you can pip-install anything, then you might as well use virtualenv.

**Comment by jayqi** [comment]
Changing this issue to consider venv _in addition_to_ and not _instead of_ virtualenv. virtualenv is still popularly used and provides a superset of the functionality of venv. However, it may make sense to include venv as an option since it is part of the Python standard library.
